queries/nim/injections.scm

```diff
@@ -0,0 +1,81 @@
+; =============================================================================
+; generalized_strings
+
+; regex in generalized_strings
+(generalized_string 
+  function: (identifier) @_string_prefix .
+  (string_content) @injection.content
+  (#set! injection.language "regex")
+  (#any-of? @_string_prefix "re" "rex"))
+
+; sql in generalized_strings
+; and anyhting you like as long as the function name is the same as the injected language's parser
+(generalized_string 
+  function: (identifier) @injection.language
+  (string_content) @injection.content)
+
+; format string in generalized_strings
+(generalized_string 
+  function: (identifier) @_string_prefix .
+  (string_content) @injection.content
+  (#set! injection.language "nim_format_string")
+  (#eq? @_string_prefix "fmt"))
```

Comment:
The second query is overlapped with the other two

```suggestion
(generalized_string 
  function: (identifier) @_string_prefix .
  (string_content) @injection.content
  (#set! injection.language "regex")
  (#any-of? @_string_prefix "re" "rex"))

; format string in generalized_strings
(generalized_string 
  function: (identifier) @_string_prefix .
  (string_content) @injection.content
  (#set! injection.language "nim_format_string")
  (#eq? @_string_prefix "fmt"))

; sql in generalized_strings
; and anything you like as long as the function name is the same as the injected language's parser
(generalized_string 
  function: (identifier) @injection.language
  (string_content) @injection.content
  (#not-any-of? @injection.language "re" "rex" "fmt"))
```

Reply:
Second thought, does injections for generalized strings with languages that's not sql, regex and fmt occur commonly? If not, I suggest to just narrow it to sql only, just like you've done for fmt and regex

Reply:
Afaik there are no other injected languages in generalized string in the stdlib. (Beside peg, which doesn't have a tree sitter parser)

But you can always define your own generalized strings and I thought it would be cool to support injection of every language just like that. You don't think it's worth it?

@alaviss what's your opinion on this?

Reply:
> You don't think it's worth it?

I don't know to be honest. It may be nice, but if it's too niche in usage, I don't think it's worth to support it. Whatever the tree-sitter-nim authors thinks will be correct, I will be going with that opinion as well.

At least, remove the overlaps for the injections. It's never nice for 2 injection queries to overlap with one another.

Reply:
yes will do. good catch

Reply:
@lucario387 

A question aside:

I have been conservative with what I put into injection queries because of
https://github.com/neovim/neovim/issues/22426
> Currently we run the full injection query on every edit to a buffer.

Technically it would be correct to support a regex like that: `f[mM][tT]` and similar for other ones. But I thought it's not worth the performance cost.
Also look at this [issue](https://github.com/aMOPel/nvim-treesitter-nim/issues/1) which shows some alternative ways to write those generalized string, for which I currently didn't include queries.

Do you think this is a good decision or should I go for more correctness and performance will be fine?

Reply:
You can add whatever you think will be better, as long as the experience between keystrokes is not 100ms long

Reply:
how can I measure that?

Reply:
(also that is just subjective testing with my hardware)

Reply:
Yes, smoothness is also subjective. Mostly, just benchmark with plenary, or eye test :D

I'm okay with performance issues for now, we can always fix it going forward